frontend/src/components/PrevNext.js

```diff
@@ -1,25 +1,71 @@
-import React from 'react';
-import Data from './data.json'
+import React, { useState, useEffect,useCallback } from "react";
+import faq from "./data.json";
+import "./styles/BodyStyle.css";
 
+const PrevNext = () => {
+  let [selectedID, setSelectedID] = useState(-1)
+  let [tab, setTab] = useState([]);
+  const toggleSelectedID = useCallback(
+    (index) => {
+      if (selectedID === index) {
+        setSelectedID(-1);
+      } else {
+        setSelectedID(index);
+      }
+    },
+    [selectedID]
+  );
 
-function PrevNext() {
-  
-    return (
-      <div>
-        <div>
-          {
-            Data.map(post => {
-              return <h4> {post.text}</h4>
-            })
-          }
-        </div>
-      </div>
-    );
-  }
-
-export default PrevNext;
-
+  useEffect(()=>{
+    let currentTab = [];
+    faq.forEach(item => {
+      currentTab.push({
+        question: item.text,
+        answer: item.answer,
+      });
+      (item.children ?? []).forEach(item1 => {
+        currentTab.push({
+          question: item1.text,
+          answer: item1.answer,
+        });
+      });
+    });
+    setTab(currentTab);
+  }, []);
```

Comment:
Do you think that is necessary as we planned to show the children data only when the user clicks on the main question button ?

In the same way I am not restructuring the json structure is very useful here as the data are already structured so we can use them, making the use of tab/currentTab obsolete. Let me know what you think